src/_tutorials/language/futures.md

```diff
@@ -59,47 +59,49 @@ Baseball score: Red Sox 10, Yankees 0
 ```
 
 Our code is problematic: since `gatherNewsReports()` blocks, the remaining
-code runs only when `gatherNewsReports()` returns with the contents of the
+code runs only after `gatherNewsReports()` returns with the contents of the
 file, _however long that takes_. And if reading the file takes a long time,
 the user waits passively, wondering if she won the lottery, what tomorrow's
 weather will be like, and who won today's game. Not good.
 
 To help keep the application responsive, Dart library authors use an
 asynchronous model when defining functions that do potentially expensive work.
-Such functions return their value using a [Future.][Future]
+Such functions return their value using a [`Future`.][Future]
 
-## What is a Future? {#what-is-a-future}
+## What is a future? {#what-is-a-future}
 
-A Future represents an asynchronous operation that returns a value (which
-can be `void`). When a function that returns a Future is invoked,
+A `Future` object represents an asynchronous operation that returns a value (which
```

Comment:
Maybe "returns" -> "produces"?

Reply:
And remove the "(which can be void)"? This little sentence has been contentious!

An unmerged PR (#1079) [changed the text](https://github.com/dart-lang/site-www/pull/1072/files#diff-197c74d48dc0011b29d95d2e3a42eba4) from "A Future represents a means for getting a value sometime in the future" to "...tracking a calculation...".

I thought that was too limiting, so when I created PR #1079, I [initially changed it](https://github.com/dart-lang/site-www/pull/1079/commits/51c8774b337803294f42c6af616495db0044a13e#diff-197c74d48dc0011b29d95d2e3a42eba4) to "A Future represents an asynchronous operation that, optionally, returns a value." We ultimately settled on the wording "... returns a value (which can be `void`)".

So... how about this:

> A future is a `Future<T>` object, which represents an asynchronous operation that produces a result of type `T`. If the result isn't a usable value, then the future's type is `Future<void>`.

Reply:
Works for me.